ecl/hqlcpp/hqlsource.cpp

```diff
@@ -3274,8 +3362,10 @@ ABoundActivity * HqlCppTranslator::doBuildActivityDiskRead(BuildCtx & ctx, IHqlE
     if (isPiped)
         return info.buildActivity(ctx, expr, TAKpiperead, "PipeRead", NULL);
     ensureDiskAccessAllowed(tableExpr);
-    if (modeOp == no_csv)
+    if ((modeOp == no_csv) && !info.genericDiskRead)
         return info.buildActivity(ctx, expr, TAKcsvread, "CsvRead", NULL);
+    if (info.genericDiskRead)
```

Comment:
Minor, but feels like code would be clearer if these two lines were moved above the modeOp==no_csv test

Reply:
changed